Sources/Testing/CMakeLists.txt

```diff
@@ -96,6 +96,11 @@ add_library(Testing
   Traits/Trait.swift)
 target_link_libraries(Testing PRIVATE
   _TestingInternals)
+if(NOT APPLE)
+  target_link_libraries(Testing PUBLIC
+    dispatch
```

Comment:
Can we optionalize this `dispatch` link as WASI doesn't support libdispatch? You can ensure the change doesn't break WASI build by `@swift-ci test WebAssembly` in cross repo testing.

Reply:
What would be the condition? `if(NOT WASI)`?

Reply:
I think it would be `if(NOT CMAKE_SYSTEM_NAME STREQUAL WASI)`.

Reply:
👍 Since `WASI` is equivalent to `CMAKE_SYSTEM_NAME STREQUAL WASI`. I went with `WASI`

Reply:
`WASI` is available only in the bleeding edge CMakes (https://gitlab.kitware.com/cmake/cmake/-/merge_requests/9659), so can we stick on `CMAKE_SYSTEM_NAME STREQUAL WASI` for now?